ndless-sdk/bin/nspire-tools

```diff
@@ -2,41 +2,58 @@
 
 function usage() {
 	echo "Usage: nspire-tools <command> <args>"
-	echo -e "   new <program>:   Generate a standard Makefile for <program>.tns"
-	echo -e "   main:            Create an empty main.c"
+	echo "   new <program>:   Generate a standard Makefile for <program>.tns"
+	echo "   main:            Create an empty main.c"
+	echo "   path:            Print out the path of the ndless installation directory"
 }
 
 
 UNAME="$(uname -s | tr A-Z a-z)"
-DIRNAME="$(dirname "$0")"
 
 # DIRNAME may not be in the toolchain directory if this script was symlinked.
 # try to find the symlink target	
-which readlink >/dev/null 2>&1 && readlink "$0" >/dev/null 2>&1 && {
-    SYMLINK="$(readlink "$0")"
-    
-    # On MacOS X, the readlink command hasn't the same behavior than the others operating systems based on Unix.
-    if [ "$UNAME" == "darwin" ]; then
-    	DIRNAME="$(dirname "$SYMLINK")"
-    else
-    	DIRNAME="${SYMLINK}/.."
-    fi
-}
+{
+  if which realpath; then
+    real0="$(realpath "$0")" || real0=$0
+  elif readlink -f /; then
+    real0="$(readlink -f "$0")" || real0=$0
+  elif which readlink; then
+    real0="$(readlink "$0")" || real0=$0
+  else
+    real0=$0
+  fi
+
+  DIRNAME="$(dirname "$real0")"
+} >/dev/null 2>&1
+
 
 # Cygwin owns a dedicated command to find the symlink target
 if [ "$UNAME" = "cygwin" ]; then
```

Comment:
Not really part of this PR, but I wonder whether this shouldn't be part of the ladder above?
Otherwise it would try to resolve the symlink target of the parent dir. I assume it was expected that `readlink` doesn't work on cygwin?

I guess noone uses cygwin anymore anyway though.

Reply:
I'm not sure exactly how it would interact, but I don't know if unix `dirname` would work on a path output by `cygpath`, so that's why I left it in that order; it still does `dirname $real0` after the ladder.

Reply:
Ne neither, but it already calls `dirname` on the output anyway, so that probably worked at some point.

Reply:
Oh, true. Maybe `SDKROOT=...` should be before the dirname? Idk, as you said, no one really uses cygwin anyway.

Reply:
Not really, as `$DIRNAME` is only used for determining `$SDKROOT`.

According to https://cygwin.com/cygwin-ug-net/cygpath.html, `--windows` means that it returns a Windows path, with `--path` being an unrelated and apparently wrong option here? It doesn't say anything about link resolution...
Assuming that the comment is correct, the intention was probably

```
elif which readlink; then
    real0="$(readlink "$0")" || real0=$0
elif  [ "$UNAME" = "cygwin" ]; then
    real0="$(cygpath "$0")" || real0=$0
else
    real0=$0
fi
```

Do you agree? Though I'd be in favor of just dropping the cygwin case here.

Reply:
Yeah, removing the cygwin code path might be best, it's incorrect and nobody's opened an issue about it, and wsl or msys2 is probably more common these days.